core-rust/core-api-server/src/core_api/conversions/lts.rs

```diff
@@ -365,13 +371,99 @@ pub fn to_api_lts_fungible_resource_balance_change(
 }
 
 pub fn to_api_lts_resultant_account_fungible_balances(
-    _database: &StateManagerDatabase,
-    _context: &MappingContext,
-    _substate_changes: &BySubstate<ChangeAction>,
+    database: &StateManagerDatabase,
+    context: &MappingContext,
+    substate_changes: &BySubstate<ChangeAction>,
 ) -> Vec<models::LtsResultantAccountFungibleBalances> {
-    // TODO(parallel PR): go through substate changes, collect fungible vaults' new values, resolve
-    // each into its global component owning the vault, retain only accounts, adjust API docs.
-    vec![]
+    let fungible_vaults = substate_changes
+        .iter_node_ids()
+        .filter(|node_id| node_id.is_internal_fungible_vault())
+        .collect::<Vec<_>>();
+    let ancestries = database
+        .batch_get_ancestry(fungible_vaults.clone())
+        .into_iter()
+        .map(|record| record.expect("InternalFungibleVault does not have a parent"));
+    let node_id_to_ancestor: HashMap<_, _> = fungible_vaults.into_iter().zip(ancestries).collect();
+    let system_db_reader = SystemDatabaseReader::new(database);
+    let mut resultant_fungible_balances_by_account = HashMap::new();
+    substate_changes
+        .iter()
+        .filter_map(|(substate_reference, change_action)| {
+            let vault_id = &substate_reference.0;
+            if !vault_id.is_internal_fungible_vault() {
+                return None;
+            }
+
+            if substate_reference.1 != MAIN_BASE_PARTITION {
+                return None;
+            }
+
+            if substate_reference.2 != FungibleVaultField::Balance.into() {
+                return None;
+            }
+
+            let account_address = node_id_to_ancestor.get(vault_id).unwrap().parent.0;
+            let Ok(account_address) = GlobalAddress::try_from(account_address) else {
+                return None;
```

Comment:
Are we sure that we will always be interested only in the global accounts? (the field name or doc does not mention "global")

E.g. I wonder how `InternalAccount`s play with this.

Reply:
As discussed on slack, `InternalAccount`s are not used in any native component and the aim of LTS is to simplify for accounts specifically.